actinia_gdi/api/gdi_ephemeral_processing_with_export.py

```diff
@@ -182,344 +120,71 @@ def post(self, location_name):
-            args.extend(additional_options)
-
-        # Export
-        p = Process(exec_type="grass",
-                         executable=module_name,
-                         executable_params=args,
-                         stdin_source=None)
-
-        self._update_num_of_steps(1)
-        self._run_module(p)
-
-    def _export_file(self, tmp_file, file_name):
-        """Export a specific file
-
-        The result is stored in a temporary directory
-        that is located in the temporary grass database.
-
-        The output file will always be compressed using zip
-
-        Args:
-            tmp_file (str): The name of the temporary file generated by a module
-            file_name (str): The file name to be used for export
-
-        Returns:
-            tuple: A tuple (file_name, output_path)
-
-        Raises:
-            AsyncProcessError: If a GRASS module return status is not 0
-
-        """
-        # Export the file
-        archive_name = file_name + ".zip"
-        # switch into the temporary working directory to use relative path for zip
-        os.chdir(self.temp_file_path)
-
-        # Compression
-        compressed_output_path = os.path.join(self.temp_file_path, archive_name)
-
-        executable = "/usr/bin/zip"
-        args = ["-r", archive_name, tmp_file]
-
-        p = Process(exec_type="exec",
-                         executable=executable,
-                         executable_params=args,
-                         stdin_source=None)
-
-        self._update_num_of_steps(1)
-        self._run_process(p)
-
-        return archive_name, compressed_output_path
-
-    def _export_resources(self, use_raster_region=False):
-        """Export all resources that were listed in the process chain description.
-
-        Save all exported files in a temporary directory first, then copy the data to its destination
-        after the export is finished. The temporary data will be finally removed.
-
-        At the moment only raster layer export is supported.
+class GdiAsyncPersistentResource(ResourceBase):
+    def __init__(self):
+        ResourceBase.__init__(self)
 
+    def post(self, location_name, mapset_name):
+        """Execute a user defined process chain in an ephemeral location/mapset and store the processing results
+        for download.
         """
-        for resource in self.resource_export_list:
-
-            # print("Check for termination %i"%self.resource_logger.get_termination(self.user_id, self.resource_id))
 
-            # Check for termination requests between the exports
-            if bool(self.resource_logger.get_termination(self.user_id, self.resource_id)) is True:
-                raise AsyncProcessTermination("Resource export was terminated by user request")
+        import pdb; pdb.set_trace()
+        # get grass and actinia module lists
+        module_list = createModuleList(self)
+        pc_list = createProcessChainTemplateList()
+        # TODO: find out size before ?
```

Comment:
Small question: Which size is meant here?

Reply:
Good question. This draft was copied from existing GdiAsyncEphemeralExportResource. I guess it wanted to avoid the execution of g.search.modules everytime a process-chain is posted to create the list of GRASS GIS modules. But while testing it and working with it, it is fast enough and I removed the TODO. (The gdi_ephemeral_processing_with_export module was rewritten in later commits).